Source/JNA/waffle-jna/src/main/java/waffle/jaas/WindowsLoginModule.java

```diff
@@ -166,6 +166,7 @@ public boolean login() throws LoginException {
                 // create the group principal and add roles as members of the group
                 final GroupPrincipal groupList = new GroupPrincipal("Roles");
                 for (final IWindowsAccount group : windowsIdentity.getGroups()) {
+                    this.principals.addAll(getRolePrincipals(group, this.roleFormat));
```

Comment:
So we take all these `role` instances and turn them into `RolePrincipal` below. This feels like duplicating entries, so feels like not the right fix. What interface is `RolePrincipal` missing vs. `Principal`? Why does your fix work?

Reply:
@dblock This is the line that was originally there and which is JAAS compliant.  The "duplication" into a GroupPrincipal (which did not exist before the breaking change), was made to get it to work with WildFly (earlier versions as it does not work with latest versions in any case).  Every Windows Group to which the User belongs, is turned into a RolePrincipal, as JAAS only knows about Users and Roles.  It does not know about Groups.  
This is why the modification to make a "custom" GroupPrincipal named "Roles" is not JAAS compliant.

I am new to gitHub and the Waffle project and I am still trying to find the ropes.  The tests failed, returning more Principals  than expected by the test routine, even for the Guest.  
I have temporarily closed the pull request until I can figure out what Windows and Windows security Groups is being returned by the test environment in order for me to modify the tests to pass.
The difference between the RolePrinciple and UserPrinciple from what I could gather is just for classification in the realm.  For example: in the Tomcat JAASRealm, you specify which classes (by name) to classify as Users and which as Roles.
The earlier modification created a seperate LIST<Principal> which it then added as a new "GroupPrincipal".  It therefore removed the RolePrinciples from the Principles and placed it into a new List<Principal> "Group"Principle, which is not understoof by Tomcat (and as a amtter of fact by any JAAS compliant realm).

Reply:
How can I see the logging output of the unit-under-test for the unit tests that have been run?